app/controllers/concerns/rails_sitemap/act_as_sitemap.rb

```diff
@@ -8,8 +8,8 @@ module ActAsSitemap
       protected
 
       def set_current_domain
-        uri = URI.parse(request.original_url)
-        pre_html = uri.html_safe? ? 'https://' : 'http://'
+        uri = URI.parse(RailsSitemap.domain || request.original_url)
+        pre_html = uri.to_s.include?('https://') ? 'https://' : 'http://'
```

Comment:
shouldn't you use just the RailsSitemap.domain url when configured without changing the protocol ?